Build (link) error with `SMS.ne4pg2_oQU480.F2010-SCREAMv1.pm-cpu_nvidia`

Noting this here.

```
/usr/bin/ld: ../../eamxx/src/mct_coupling/libscream_theta-l_kokkos_4_128_10.a(homme_grid_mod.F90.o): in function `cleanup_grid_init_data_f90':
/global/cfs/cdirs/e3sm/ndk/repos/c31-nov6/components/eamxx/src/dynamics/homme/interface/homme_grid_mod.F90:86:(.text+0x1cc): undefined reference to `check_grids_inited_'
/usr/bin/ld: ../../eamxx/src/mct_coupling/libscream_theta-l_kokkos_4_128_10.a(homme_grid_mod.F90.o): in function `finalize_geometry_f90':
/global/cfs/cdirs/e3sm/ndk/repos/c31-nov6/components/eamxx/src/dynamics/homme/interface/homme_grid_mod.F90:98:(.text+0x20c): undefined reference to `check_grids_inited_'
```

--- a/components/eamxx/src/dynamics/homme/interface/homme_grid_mod.F90
+++ b/components/eamxx/src/dynamics/homme/interface/homme_grid_mod.F90
@@ -12,11 +12,12 @@ module homme_grid_mod
   public :: finalize_geometry_f90
 
   ! Routines to get information
-  ! public :: get_cols_gids_f90, get_cols_indices_f90
   public :: get_num_local_columns_f90, get_num_global_columns_f90
   public :: get_num_local_elems_f90, get_num_global_elems_f90
   public :: get_np_f90, get_nlev_f90
   public :: is_planar_geometry_f90
+
+  public :: check_grids_inited
 
 contains
 

